components/governance/org.wso2.carbon.governance.rest.api/src/main/java/org/wso2/carbon/governance/rest/api/Asset.java

```diff
@@ -449,10 +457,26 @@ private Response createGovernanceAsset(String assetType, DetachedGenericArtifact
         String shortName = Util.getShortName(assetType);
         if (validateAssetType(shortName)){
 
-            if(isContentType(shortName , genericArtifact)){
-                //TODO
-                Response.status(Response.Status.CONFLICT).build();
-                //createContentAsset(shortName, genericArtifact);
+            if (isContentType(shortName, genericArtifact)) {
+                String[][] propertyArray = new String[2][2];
+                propertyArray[0][0] = "version";
+                propertyArray[0][1] = genericArtifact.getAttribute(OVERVIEW_VERSION);
+                propertyArray[1][0] = "resource.source";
+                propertyArray[1][1] = "GovernanceAPI";
+
+                Registry registry = getUserRegistry();
+
+                // overview_path is not a mandatory for known asset types.
+                // for unknown asset types '/_system/governance/' will be appended by the system
+                boolean isSuccessful = importResourceWithRegistry(registry,
+                        genericArtifact.getAttribute("overview_path") + genericArtifact.getAttribute(OVERVIEW_VERSION),
```

Comment:
As per the implementation overview_path means storage path of the asset. This is not a mandatory input for known artifact types. I have added java docs for the importResourceWithRegistry() method to increase the readability. Kindly find the sample request below for uploading a wsdl.

{
  "name":"testwsdl1",
  "version":"1.0.0",
  "type":"application/wsdl+xml",
  "url":"http://www.webservicex.net/CovertPressure.asmx?WSDL",
  "path": ""
}

for unknown artifact types 

{
  "name":"testNew1",
  "version":"1.0.0",
  "type":"application/new+xml",
  "url":"http://www.host.net/unkownFileType.new",
  "path": "/trunk/new/storage/path/"
}